age/keysource_test.go

```diff
@@ -380,8 +380,6 @@ func overwriteUserConfigDir(t *testing.T, path string) {
 	switch runtime.GOOS {
 	case "windows":
 		t.Setenv("AppData", path)
-	case "darwin", "ios": // This adds "/Library/Application Support" as a suffix to $HOME
```

Comment:
Would maintain this, and add an individual test for `getUserConfigDir` to ensure the custom `darwin` behavior works as expected.

Reply:
I added a test. I cannot run it on macOS though, only on Linux.

Reply:
You added a test for `overwriteUserConfigDir`, and not for `getUserConfigDir` :smile:

Reply:
Well, it's testing both. To test `getUserConfigDir()` we somehow need to set the correct environment variable. And there's already a helper for doing that, called `overwriteUserConfigDir`, so why not use that one instead of repeating the same code?

Reply:
But we now do not appear to test the Darwin specific behavior around preferring the environment variable if set?

Reply:
We do: on Darwin `overwriteUserConfigDir` now sets `XDG_CONFIG_HOME` (`default` case).

Reply:
Ah _doh_, I am suffering from broken nights :-).

I think maintaining the logic in `overwriteUserConfigDir` and adding a specific test case for `XDG_CONFIG_HOME` on Darwin (testing `getUserConfigDir`) results in better maintainable tests and is easier to grasp (~for example, for people with limited sleep~), in which the test case is descriptive for the feature you are adding. Instead of it being indirectly tested via a certain construction.

Reply:
I adjusted the test to be more explicit. I didn't restore the `overwriteUserConfigDir` behavior though since it would now be incorrect if the user has `XDG_CONFIG_HOME` set.